application/src/main/kotlin/net/corda/v5/application/marshalling/JsonMarshallingService.kt

```diff
@@ -0,0 +1,32 @@
+package net.corda.v5.application.marshalling
+
+import net.corda.v5.base.annotations.DoNotImplement
+
+/**
+ * An optional service CorDapps and other services may use to marshall arbitrary content in and out of JSON format using standard/approved
+ * mappers.
+ */
+@DoNotImplement
+interface JsonMarshallingService : MarshallingService {
+
+    /**
+     * Deserializes the [input] JSON into a list of instances of [T].
+     *
+     * @param input The JSON to deserialize.
+     * @param clazz The [Class] type to deserialize into.
+     *
+     * @return A new list of [T].
+     */
+    fun <T> parseList(input: String, clazz: Class<T>): List<T>
```

Comment:
I left it off as it's not clear to me it will always be possible to implement it. I think it's there on the `JsonMarshallingService` because constructing the List `clazz` is non-trivial (the implementation reaches into the object mapper to do it). I think it not always being possible to implement is a bit of a weak argument though, as we could probably work around by e.g. returning a list with a single item in for cases where it doesn't make sense.

Reply:
I have a feeling we need this moved up to `MarshallingService`.  Otherwise I don't think we'll be able to have list types as the top level JSON type as I think you have to tell Jackson what you're expecting for security?

i.e. probably need a `getRequestBodyAsList()` that uses it.

Worth tests to see if that is so?

Begs the question, what about Map (or is that not a JSON thing)?  But okay to leave that for later, definitely.

Reply:
If we want a `getRequestBodyAsList()` then we probably need to move this down a level regardless. I'll have a look at that this afternoon.

For wider collections I think Jackson has the same thing, it's just we only implemented it for lists before. That would need a little more thought on the implementation side I think.